cmd/clusterctl/examples/openstack/centos/provider-components.yaml.template

```diff
@@ -19,204 +19,19 @@ data:
   clouds.yaml: $OPENSTACK_CLOUD_CONFIG
 ---
 apiVersion: v1
-kind: ConfigMap
+kind: Secret
+type: Opaque
+metadata:
+  name: worker-user-data
+  namespace: openstack-provider-system
+data:
+  userData: $WORKER_USER_DATA
+---
+apiVersion: v1
+kind: Secret
+type: Opaque
 metadata:
-  name: machine-setup
+  name: master-user-data
```

Comment:
This comment doesn't make much sense here... but you forgot to edit `examples/openstack/centos/machines.yaml.template` and add `userDataSecret` (https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/7ff2fcb1c095db35d2701a1fa12658869a249633/cmd/clusterctl/examples/openstack/centos/machines.yaml.template)